### PR TITLE
Milestone 3: Set up associations.

### PR DESCRIPTION
**Pull request by dfsalam**
# In this pull request I:

### Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental.

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# Class that represents a rental
+class Rental
+  attr_reader :person, :book
+
+  def initialize(person, book)
+    @person = person
+    @book = book
+    person.add_rental(self)
+    book.add_rental(self)
+  end
```
Comment (TadesseDev) [inline rental.rb]: - [ ] Dear @dfsalam, while creating a rental we need to set the date for the rental happens, Please make sure to have a rental variable that gen initialized while initializing the new rental, Also, please make sure to have the `setters` and `getters` for the `@date` instance variable.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+# Class that represents a book
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
```
Comment (TadesseDev) [inline book.rb]: - [ ] Good job adding the getter and setter for `title` and `author`. Please also set the Getters and Setters for `@rentals`. users should be able to get and set the rental of the book.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @dfsalam 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x]  Correct GitFlow  ✅
 - [x]  Good commit history  ✅
 - [x]  Good PR title/description  ✅
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by CollinsTatang — COMMENTED** [review]
Hi @dfsalam, 

While you made a great effort in this project. Unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted the wrong Pull Request to this project.

- [ ] Please check your dashboard again and resubmit the right Pull Request.



Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by TadesseDev — APPROVED** [review]
### Hi @dfsalam 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [x] Good job making the changes ✅
 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**